include/hipSYCL/sycl/libkernel/backend.hpp

```diff
@@ -62,6 +62,7 @@
 
 #ifdef HIPSYCL_LIBKERNEL_DEVICE_PASS
  #define SYCL_DEVICE_ONLY
+ #define __SYCL_DEVICE_ONLY__
```

Comment:
Only define `__SYCL_DEVICE_ONLY__` if it's not already defined. When compiling with DPC++ for SPIR-V, DPC++ will already handle it.

Reply:
Ahh yeah you told me that, sorry